pkg/detector/detector.go

```diff
@@ -1039,20 +1039,20 @@ func (d *ResourceDetector) HandlePropagationPolicyDeletion(policyNS string, poli
 	}
 
 	for index, binding := range rbs.Items {
-		// Cleanup the labels from the reference binding so that the karmada scheduler won't reschedule the binding.
-		if err := d.CleanupResourceBindingLabels(&rbs.Items[index], policyv1alpha1.PropagationPolicyNamespaceLabel, policyv1alpha1.PropagationPolicyNameLabel); err != nil {
-			klog.Errorf("Failed to cleanup label from resource binding(%s/%s) when propagation policy(%s/%s) removing, error: %v",
-				binding.Namespace, binding.Name, policyNS, policyName, err)
-			return err
-		}
-
-		// Cleanup the labels from the object referencing by binding.
+		// Clean up the labels from the object referencing by binding first to ensure idempotence.
```

Comment:
We may not need to emphasize idempotence here, as it is not easy to understand. We only need to explain why we need to do that. Something like:
```
		// Must remove the labels from the resource template ahead of ResourceBinding, otherwise might lose the chance
		// to do that in a retry loop(in particular, the label was successfully removed from ResourceBinding, but
		// resource template not), since the ResourceBinding will not be listed again.
```

Reply:
done